services/121-service/src/payments/fsp-integration/safaricom/safaricom.service.ts

```diff
@@ -157,13 +95,13 @@ export class SafaricomService
       InitiatorName: process.env.SAFARICOM_INITIATORNAME!,
       SecurityCredential: process.env.SAFARICOM_SECURITY_CREDENTIAL!,
       CommandID: 'BusinessPayment',
-      Amount: paymentData.transactionAmount,
+      Amount: transactionAmount,
       PartyA: process.env.SAFARICOM_PARTY_A!,
-      PartyB: paymentData.paymentAddress,
+      PartyB: phoneNumber, // Set to empty string to trigger mock failure
```

Comment:
Refactor: we should not let the mock fail on empty string, since an empty phone number will be (or should be) called earlier, resulting in a failed transaction. Maybe do something like: +3100000000?

Reply:
processed